### DISCUSSION TITLE
I want to make two api calls and merge the response to show

### DISCUSSION THREAD

**Discussion by shahzadhemani17** [Budibase/budibase#5513] [Help]
Normally in design we provide api in data source and in transformer we do some data processing, But in my case, i want to send two api calls and process data to make one response. Can anyone guide me with this use case?

**Comment by shahzadhemani17** [comment]
Here you can see i am just making a function async and there is no response, this code is getting executed in transformer  




```
// Start function
const start = async function(a, b) {
  return { "one":1, "two":2, "three":3, "four":4, "five":5 };
  
}

// Call start
return  start();
```

**Comment by melohagan** [comment]
Hi @shahzadhemani17 

Async functions are not supported. 

What you could try is chaining executed queries: https://docs.budibase.com/docs/referencing

 1. Execute Query
 2. Execute Query
 3. Action that uses a combination of the results from action 1 & 2

**Reply by cngi1** [reply]
@melohagan Not to hijack this but could you please show how step 3 could be done.  My main issue would be how to reference data from the two queries in the action.  I would expect the 2nd query must be showing right shift under the first query to have the references for the action right? 
Thank you

**Reply by mjashanks** [reply]
@cngi1 you can use the result in the binding of subsequent actions as below:

![image](https://user-images.githubusercontent.com/3524181/164747971-efcb67e2-7a21-427c-abc7-4d95e1e6aaaa.png)
